helix-term/src/ui/completion.rs

```diff
@@ -92,8 +92,10 @@ impl menu::Item for CompletionItem {
                                 value, ..
                             }) => value,
                         };
-                        Color::from_hex(text)
+                        // LSPs which send Color completion items include a 6 digit hex code at the end for the color. The extra 1 digit is for the '#'
+                        text.get(text.len() - 7..)
```

Comment:
This subtraction can underflow if a language server sends a documentation field shorter than 7 bytes. If it's shorter than 7 bytes then we can return early anyways since it's not long enough to contain a hex code:

```suggestion
                        text.get(text.len().checked_sub(7)?..)
```

Reply:
Thanks, good catch!